### PR TITLE
feat(linter): implement eslint-plugin-unicorn/no-thenable rule

### PR DESCRIPTION
**Pull request by Devin-Yeung**
related to #684

### CODE REVIEW

**Comment by Devin-Yeung** [comment]
It's almost done, I will find some time to:

- [ ] fix clippy warning
- [ ] write doc

File: crates/oxc_linter/src/rules/unicorn/no_thenable.rs [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+            AstKind::MemberExpression(expr) => {
+                if let Some(parent) = ctx.nodes().parent_node(node.id()) {
+                    if let Some(grandparent) = ctx.nodes().parent_node(parent.id()) {
+                        if let AstKind::AssignmentTarget(
+                            AssignmentTarget::SimpleAssignmentTarget(
+                                SimpleAssignmentTarget::MemberAssignmentTarget(target),
+                            ),
+                        ) = grandparent.kind()
+                        {
+                            if hash(&expr) == hash(target) {
```
Comment (Devin-Yeung) [inline crates/oxc_linter/src/rules/unicorn/no_thenable.rs:193]: not sure if this is the preferred way to compare two AST 🤔

Comment (mysteryven) [reply crates/oxc_linter/src/rules/unicorn/no_thenable.rs:193]: Seems we don't need to check this. Or do you think if we can match `AssignmentTarget::SimpleAssignmentTarget` first?

Comment (Devin-Yeung) [reply crates/oxc_linter/src/rules/unicorn/no_thenable.rs:193]: > Seems we don't need to check this. Or do you think if we can match `AssignmentTarget::SimpleAssignmentTarget` first?

Thank you for pointing it out, I will give a try

**Review by Boshen — APPROVED** [review]

**Pull request merged**